В классе StudentController cоздан эндпоинт GET /students/print-parall…

Manikgog opened the pull request:
…el. Эндпоинт GET /students/print-parallel обращается к сервису, где первые 2 студента распечатываются в Main потоке, а 4 следующих имени распечатываются в двух других потоках.

В классе StudentController cоздан эндпоинт  GET /students/print-synchronized Создан синхронизированный метод для вывода имен студентов в консоль. Эндпоинт GET /students/print-synchronized обращается к сервису, где первые 2 студента распечатываются в Main потоке, а 4 следующих имени распечатываются через синхронизированный метод.

Manikgog committed:
В классе StudentController cоздан эндпоинт GET /students/print-parallel. Эндпоинт GET /students/print-parallel обращается к сервису, где первые 2 студента распечатываются в Main потоке, а 4 следующих имени распечатываются в двух других потоках.
В классе StudentController cоздан эндпоинт  GET /students/print-synchronized Создан синхронизированный метод для вывода имен студентов в консоль. Эндпоинт GET /students/print-synchronized обращается к сервису, где первые 2 студента распечатываются в Main потоке, а 4 следующих имени распечатываются через синхронизированный метод.

dmitry-bizin commented on src/main/java/ru/hogwarts/school/service/StudentService.java:
```diff
@@ -134,4 +134,35 @@ public String getLongestName() {
     }
+
+    public void printStudentsInParallel() {
+        List<Student> listOfStudents = studentRepository.findAll().stream().limit(6).toList();
```
лучше из базы сразу достать 6, чем достать всех, а потом оставить 6

dmitry-bizin commented on src/main/java/ru/hogwarts/school/service/StudentService.java:
```diff
@@ -134,4 +134,35 @@ public String getLongestName() {
+    }
+
+    public void printStudentsInParallelSynchronized() {
+        List<Student> listOfStudents = studentRepository.findAll().stream().limit(6).toList();
```
лучше из базы сразу достать 6, чем достать всех, а потом оставить 6

Manikgog committed:
В интерфейсе StudentRepository добавлен метод List<Student> getFirstNStudents(int count); для получения 6 студентов.
В классе StudentService исправлены методы void printStudentsInParallel() и void printStudentsInParallelSynchronized(), которые теперь используют метод  getFirstNStudents(int count).

Manikgog committed:
В интерфейсе StudentRepository добавлен метод List<Student> getFirstNStudents(int count); для получения 6 студентов.
В классе StudentService исправлены методы void printStudentsInParallel() и void printStudentsInParallelSynchronized(), которые теперь используют метод  getFirstNStudents(int count).
